src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -67,48 +58,13 @@ const TopicFragment = ({
     }));
   }, [unfollowedTopic]);
 
-  React.useEffect(() => {
-    const parseToken = async () => {
-      const id = await getUserId();
-      if (id) {
-        setMyId(id);
-      }
-    };
-    parseToken();
-  }, []);
   const handleScroll = React.useCallback(() => {
     Keyboard.dismiss();
   });
 
-  const exhangeFollower = (newTopicLists, willFollow, topicId) => {
-    const indexTopic = newTopicLists.findIndex((item) => item.topic_id === topicId);
-    newTopicLists[indexTopic].following = !!willFollow;
-    // newTopicLists[indexTopic].user_id_follower = myId;
-    return newTopicLists[indexTopic];
-  };
-
-  const mapTopic = (newTopic) => {
-    return discovery.initialTopics.map((topic) => {
-      if (topic.topic_id === newTopic.topic_id) {
-        return newTopic;
-      }
-      return topic;
-    });
-  };
-
-  const handleTopic = (from, willFollow, item, index) => {
-    if (from === FROM_FOLLOWED_TOPIC_INITIAL) {
-      const newFollowedTopics = [...topics];
-      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
-
-      DiscoveryAction.setDiscoveryInitialTopics(mapTopic(newTopic), discoveryDispatch);
-    }
-
-    if (from === FROM_UNFOLLOWED_TOPIC_INITIAL) {
-      const newFollowedTopics = [...topics];
-      const newTopic = exhangeFollower(newFollowedTopics, willFollow, item.topic_id);
-
-      DiscoveryAction.setDiscoveryInitialTopics(mapTopic(newTopic), discoveryDispatch);
+  const handleTopic = (from, willFollow, item) => {
+    if (from === FROM_FOLLOWED_TOPIC_INITIAL || from === FROM_UNFOLLOWED_TOPIC_INITIAL) {
+      updateFollowTopicDiscoveryContext(willFollow, item);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change seems to simplify the logic of the `handleTopic` function by removing some redundant code. It's crucial to ensure that the `updateFollowTopicDiscoveryContext` function is implemented correctly and updates the state as expected.



<!-- This is an auto-generated comment by OSS CodeRabbit -->